src/selectors.js

```diff
@@ -2,8 +2,18 @@ import { matchPath } from "react-router"
 
 const createSelectors = (structure) => {
   const { getIn, toJS } = structure
-  const getLocation = state => toJS(getIn(state, ['router', 'location']))
-  const getAction = state => toJS(getIn(state, ['router', 'action']))
+
+  const _getLocation = state => getIn(state, ['location'])
+  const _getAction = state => getIn(state, ['action'])
+  const isRouter = (value) => value != null && typeof value === 'object' && _getLocation(value) && _getAction(value)
```

Comment:
@supasate Do you think there's value in doing this check to make sure the right object shape is mounted?

Reply:
Yes. I think it's good to check :)